### PR TITLE
Feature/visit screen

### PR DESCRIPTION
**Pull request by samip77**

### CODE REVIEW

**Comment by dipkasyap** [comment]
why the latest build is failing?

File: Covid Track/Covid Track/Models/Visitor/VisitorModel.swift [hunk 1]
```diff
@@ -7,7 +7,60 @@
...
+//    public init(id: String , name: String, email: String, phone: String, homeAddress: String?, workAddress: String?) {
+//        self.id = id
+//        self.name = name
+//        self.email = email
+//        self.phone = phone
+//        self.homeAddress = homeAddress
+//        self.workAddress = workAddress
+//    }
+//
+//}
```
Comment (dipkasyap) [inline Covid Track/Covid Track/Models/Visitor/VisitorModel.swift:11]: Why commented codes are in pull request?

File: Covid Track/Covid Track/Utilities/Extension/Views/Cell+ResubaleView.swift [hunk 2]
```diff
@@ -10,11 +10,11 @@ import Foundation
...
 
 protocol ReusableView {
-  static var defaultResuseIdentifier: String { get }
+  static var defaultReuseIdentifier: String { get }
 }
 
 extension ReusableView where Self: UIView {
-  static var defaultResuseIdentifier: String {
+  static var defaultReuseIdentifier: String {
     return NSStringFromClass(self)
```
Comment (dipkasyap) [inline Covid Track/Covid Track/Utilities/Extension/Views/Cell+ResubaleView.swift:18]: Please follow the proper naming convention.

File: Covid Track/Covid Track/ViewModels/Visit History/VisitHistoryViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+//
+
+import Foundation
+
+class VisitHistoryViewModel {
+  
+  private let visitRepository:VisitRepository
+  
+  var visits:[VisitViewModel] = []
+
```
Comment (dipkasyap) [inline Covid Track/Covid Track/ViewModels/Visit History/VisitHistoryViewModel.swift:16]: Why store property is accessible without setter and getter?

File: Covid Track/Covid Track/ViewModels/Visit History/VisitViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+  
+  private func readableTimeFrom(date: Date) -> String{
+    let timeInterval = Date().timeIntervalSince(date)
+    var description = ""
+   
+    let second = 1.0
+    let min = 60 * second
+    let hour = 60 * min
+    let day = 24 * hour
+    let year = 365 * day
```
Comment (dipkasyap) [inline Covid Track/Covid Track/ViewModels/Visit History/VisitViewModel.swift:102]: Use proper setter and getter as per required

File: Covid Track/Covid Track/ViewModels/Visit Record/VisitRecordViewModel.swift [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+//
+//  Created by Samip shah on 28/7/20.
+//  Copyright © 2020 Samip shah. All rights reserved.
+//
+
+import Foundation
+
+class VisitRecordError:Error {
+  var localizedDescription: String
+
```
Comment (dipkasyap) [inline Covid Track/Covid Track/ViewModels/Visit Record/VisitRecordViewModel.swift:13]: why localizedDescription is publically assinable?

File: Covid Track/Covid Track/Views/Controllers/Visit Record/VisitRecordViewController.swift [hunk 6]
```diff
@@ -8,23 +8,175 @@
...
+  @IBOutlet weak var emailTextField: UITextField!
+  @IBOutlet weak var phoneTextField: UITextField!
+  @IBOutlet weak var homeTextField: UITextField!
+  @IBOutlet weak var workTextField: UITextField!
+  
+  @IBOutlet weak var datePicker: UIDatePicker!
+  @IBOutlet weak var temperaturePicker: UIPickerView!
+  @IBOutlet weak var profileButton: UIButton!
+  @IBOutlet weak var moreInfoButton: UIButton!
+  @IBOutlet weak var checkInButton: UIButton!
```
Comment (dipkasyap) [inline Covid Track/Covid Track/Views/Controllers/Visit Record/VisitRecordViewController.swift]: if possible make IBOutlet private.

**Review by dipkasyap — CHANGES_REQUESTED** [review]

File: Covid Track/Covid Track/Models/Visitor/VisitorModel.swift (hunk 1 above)
Comment (samip77) [reply Covid Track/Covid Track/Models/Visitor/VisitorModel.swift:11]: Thanks. I have fixed it now.

File: Covid Track/Covid Track/ViewModels/Visit History/VisitHistoryViewModel.swift (hunk 3 above)
Comment (samip77) [reply Covid Track/Covid Track/ViewModels/Visit History/VisitHistoryViewModel.swift:16]: I have made the visit property private as it needs to accessed inside the class only.

File: Covid Track/Covid Track/Utilities/Extension/Views/Cell+ResubaleView.swift (hunk 2 above)
Comment (samip77) [reply Covid Track/Covid Track/Utilities/Extension/Views/Cell+ResubaleView.swift:18]: Apple uses `open func register(_ cellClass: AnyClass?, forCellReuseIdentifier identifier: String)` function to register tableview cell. I have updated name variable name to - reuseIdentifier

File: Covid Track/Covid Track/ViewModels/Visit History/VisitViewModel.swift (hunk 4 above)
Comment (samip77) [reply Covid Track/Covid Track/ViewModels/Visit History/VisitViewModel.swift:102]: Those are the constants used for calculating time ago, and their scope is inside a function only, so I believe it doesn't need to be changed.

File: Covid Track/Covid Track/ViewModels/Visit Record/VisitRecordViewModel.swift (hunk 5 above)
Comment (samip77) [reply Covid Track/Covid Track/ViewModels/Visit Record/VisitRecordViewModel.swift:13]: Access modifier has been changed to private set.

File: Covid Track/Covid Track/Views/Controllers/Visit Record/VisitRecordViewController.swift (hunk 6 above)
Comment (samip77) [reply Covid Track/Covid Track/Views/Controllers/Visit Record/VisitRecordViewController.swift]: Done.

**Pull request merged**